iroha_http_server/src/lib.rs

```diff
@@ -0,0 +1,640 @@
+        if version == 1 {
+            Ok(HttpVersion::Http1_1)
+        } else {
+            Err("Http version not supported.".to_string())
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct HttpRequest {
+    /// The request method, such as `GET`.
+    pub method: String,
+    /// The request path, such as `/about-us`.
+    pub path: String,
+    /// The request version, such as `HTTP/1.1`.
+    pub version: HttpVersion,
+    /// The request headers. Keys will be all lowercase for parsing simplicity.
+    pub headers: Headers,
+    /// The request body in binary form.
+    pub body: Vec<u8>,
+}
+
+impl<'h, 'b> TryFrom<HttpParseRequest<'h, 'b>> for HttpRequest {
+    type Error = String;
+
+    fn try_from(request: HttpParseRequest<'h, 'b>) -> Result<Self, Self::Error> {
+        Ok(HttpRequest {
+            method: request.method.ok_or("Method not found.")?.to_string(),
+            path: request.path.ok_or("Path not found.")?.to_string(),
+            version: request.version.ok_or("Version not found.")?.try_into()?,
+            headers: request
+                .headers
+                .iter()
+                .map(|header| (header.name.to_lowercase(), header.value.to_vec()))
+                .collect(),
+            body: Vec::new(),
+        })
+    }
+}
+
+impl TryFrom<&[u8]> for HttpRequest {
+    type Error = String;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        let mut headers = [httparse::EMPTY_HEADER; MAX_HEADERS];
+        let mut request = httparse::Request::new(&mut headers);
+        if let Status::Complete(header_size) =
+            request.parse(&bytes).map_err(|err| err.to_string())?
+        {
+            let mut request: HttpRequest = request.try_into()?;
+            //TODO: Deal with chunked messages which do not have Content-Length header
+            //They instead have Transfer-Encoding: `Chunked` https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.4
+            if let Some(content_length) = request.headers.get(&CONTENT_LENGTH_HEADER.to_lowercase())
+            {
+                let content_length: usize = String::from_utf8(content_length.clone())
+                    .map_err(|_| "Failed to parse content length value - invalid utf-8.")?
+                    .parse()
+                    .map_err(|_| "Failed to parse content length value - not a number.")?;
+                request.body = bytes[header_size..(header_size + content_length)].to_vec();
+            }
+            Ok(request)
+        } else {
+            Err("Failed to read header.".to_string())
+        }
+    }
+}
+
+//TODO: Add more response builders.
+#[derive(Debug, Clone, PartialEq)]
+pub struct HttpResponse {
+    /// The response version, such as `HTTP/1.1`.
+    pub version: HttpVersion,
+    /// The response code, such as `200`.
+    pub code: u16,
+    /// The response reason-phrase, such as `OK`.
+    pub reason: String,
```

Comment:
May be replaced with enums?

Reply:
The reason phrase is not fixed as specified in 6.1.1 section of the [standard](https://www.w3.org/Protocols/rfc2616/rfc2616-sec6.html). So the server might want to use a custom phrase, therefore I think it is better to leave it as `String`.

As for the code itself I think we don't want to waste time specifying all the possible codes at the moment, but the library user might want to use some not very common response, so I also think we should leave it as it is. At least for now.